### ISSUE TITLE
GDPR consent problem: PowerShell 7.4.5 installer forcefully enables telemetry on upgrade even if previous version was installed with telemetry disabled

### ISSUE THREAD

**Issue by R-Adrian** [PowerShell/PowerShell#21467]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

- if an existing version of PowerShell 7 is installed then uninstall it from control panel.
- make sure there is no C:\Program Files\PowerShell\7 folder left 
- download and install the immediately previous version
(e.g. 7.4.1 in our case here `https://github.com/PowerShell/PowerShell/releases/download/v7.4.1/PowerShell-7.4.1-win-x64.msi` )
- install that version. When installing make sure to enable the checkbox for opt out, to disable telemetry.
- start pwsh, check the output of the `$env:POWERSHELL_TELEMETRY_OPTOUT` environment variable
(it shows 1 in my case)

for testing silent / unattended upgrade - make sure pwsh 7 is not running, start an elevated cmd.exe command prompt, run `winget upgrade Microsoft.PowerShell`  - and update pwsh to version 7.4.2 ( `https://github.com/PowerShell/PowerShell/releases/download/v7.4.2/PowerShell-7.4.2-win-x64.msi` )

for testing interactive upgrade: run the MSI installer manually - the same result can be observed - the telemetry preference is reset to defaut (i.e. the checkbox is disabled in the MSI installer running an upgrade).

- after the upgrade start pwsh and check the output of that environment variable again - after the winget silent/unattended update it shows no optout (i.e. the telemetry is now forcefully enabled again)

### Expected behavior
telemetry remains disabled

### Actual behavior
telemetry is forcefully enabled for either interactive updates (the opt out preference is not used by the MSI installer) or silent / unattended updates

### Error details
installer issue

### Environment data
(MSI installer issue - env not really relevant)

### Visuals

PowerShell telemetry reference:
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_telemetry?view=powershell-7.4

PowerShell 7.4.1 manual installation of the MSI file, with telemetry disabled:
![PowerShell 7.4.1 installation with telemetry disabled](https://github.com/microsoft/winget-pkgs/assets/16319504/2ea4ee93-b6a3-48fe-9757-5af6cdccf15e)

checking before upgrade...
![pwsh_741_before_upgrade](https://github.com/microsoft/winget-pkgs/assets/16319504/0a8eb02c-ddf0-4999-8707-f99e224c302c)

and running  `winget upgrade  Microsoft.PowerShell` (using the old Windows PowerShell instead of cmd.exe as a console window)

![pwsh_742_winget_upgrade_](https://github.com/microsoft/winget-pkgs/assets/16319504/083d141b-683c-4dca-8629-0c62aa0f189f)

... PowerShell 7 telemetry optout is now disabled and the variable is not defined anymore - this means that telemetry was forcefully enabled during the upgrade.

![pwsh_742_after_upgrade](https://github.com/microsoft/winget-pkgs/assets/16319504/76a39352-dfe7-4b74-b9fe-ee516bf26216)

**Comment by StevenBucher98** [comment]
Additional context and information in this issue: https://github.com/microsoft/winget-pkgs/issues/

**Comment by R-Adrian** [comment]
note: that the issue opened for WinGet was closed today with the reason "This issue has been identified as requiring a fix from a third party or external repository. Since there has been no recent activity on this issue, it will be automatically closed."

Since pull request https://github.com/PowerShell/PowerShell/pull/20420 has been merged but no release based on it is yet available i will keep this one open until that fix gets released via Windows Update.


and a side note: i think because of that enforced mandatory consent issue / bug, all PowerShell telemetry, both collected in the past and the future collected telemetry can now be considered legally compromised unless there is a clearly defined `$env:POWERSHELL_TELEMETRY_OPTOUT = 0` environment variable

i.e. the simple lack of the optout environment variable can no longer be considered as informed consent, on the contrary it is highly suspect now because the user might have opted out when they installed PowerShell 7 but the version updates since then have wiped out that preference.

**Comment by StevenBucher98** [comment]
@R-Adrian it is available in 7.5.0-preview.3, can you see if this still reproduces for you?

**Comment by R-Adrian** [comment]
i see that v7.4.3 is now published via Windows Update....  (published on July 22nd, 2024) and that version contains the pull request that *technically* fixes the inheritance of settings from previous version, including telemetry consent preference...

https://www.catalog.update.microsoft.com/Search.aspx?q=powershell%20v7.4.3
https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=b1ee1069-f9aa-4f67-b637-b5d26239e0c7


I think that the technical issue can be closed ... but it still remains as a LEGAL issue - because of that problem the legal basis for consent for every telemetry optout that is missing in Europe is now compromised and Microsoft could be considered in violation of the GDPR - multiple violations in fact, for each source instance of telemetry collected.

I think that the proper fix for future versions would be to include additional messages / prompts for consent to be shown next time that pwsh is started in interactive mode (not for scripts though, that would mess scripting)... and if it is running with administrative rights save the new consent preference in a permanent environment variable.

**Comment by R-Adrian** [comment]
**re-opening the issue:**
i see that PowerShell version 7.4.4 was released yesterday https://github.com/PowerShell/PowerShell/releases/tag/v7.4.4
and it seems to have a regression on this: it does not preserve anymore the initial installation settings from the previous versions (and also wipes the telemetry optout too) - it wants to install with telemetry enabled.

i used the file:
PowerShell-7.4.4-win-x64.msi  SHA-256: c755a79759ad5daa5f76a855abb46be1b9ce86616607138fef5b02ec4bfae643

0) i have uninstalled any pwsh v7 versions that i had... made sure the C:\Program Files\PowerShell was deleted and then...

1) re-installed v7.4.3 with the MSI from GitHub, made sure to tick the checkbox for telemetry optout in the 7.4.3 installer,
Verified that the $env:POWERSHELL_TELEMETRY_OPTOUT variable is present and defined when starting both pwsh and the classic Windows PowerShell
..also verified that the 7.4.4 MSI installer ignores again the telemetry optout when running before installing pwsh from MS Update Catalog (see details below)

2) installed the same v7.4.3 downloaded from the Windows Update catalog
File: powershell-7.4.3-win-x64_54091441f1ce659f9e6e493fca74c359265d819f.exe
 SHA-256: 168276b247543356bb689001041ee82e662c68fa9de74cb34c5392af98219343
verified that the $env:POWERSHELL_TELEMETRY_OPTOUT variable is still present and defined after installing the v7.4.3 from MS Catalog on top of the 7.4.3 MSI downloaded from here on GitHub.
(the installer from MS Update Catalog has preserved the setting but has created a duplicate add-remove programs entry - will open a separate issue for that)

3) ... and when i start the 7.4.4 installer to upgrade, it ignores the telemetry optout preference already defined by the 7.4.3 MSI installer and wants to forcefully enable telemetry again. 

![image](https://github.com/user-attachments/assets/2b40c0df-fbe0-4a96-b5c9-6340d50c47d4)

**Comment by StevenBucher98** [comment]
Sharing here that I confirmed this seems to be only reproducable via winget, other non-interactive ways like using msiexec to upgrade respects the properties set.

**Comment by denelon** [comment]
@StevenBucher98 I'd be more than happy to collaborate to figure out what we can do here. Maybe there is a switch we can pass to the installer or something else to address this issue.

**Comment by R-Adrian** [comment]
just noticed that pwsh 7.4.5 is now available and it supposedly now preserves the installation options from the previous version...

https://github.com/PowerShell/PowerShell/pull/24143

unfortunately that is still not the case, i just upgraded from 7.4.4 to 7.4.5 via winget and the 7.4.5 installer forcefully enabled telemetry yet again, by wiping the environment optout.

screenshot before installation:
![image](https://github.com/user-attachments/assets/b190a2e6-5544-4c7e-bc79-008687daa24b)


i closed pwsh v7.4.4 and ran the winget upgrade... then...
![image](https://github.com/user-attachments/assets/f4d30a17-6b76-4bd2-b64c-c0987aad5207)


q.e.d. the telemetry optout was wiped again by the v7.4.5 installer,

**Comment by R-Adrian** [comment]
winget started showing today that pwsh version 7.4.6 is available as upgrade from 7.4.5

before upgrade:
```
> $env:POWERSHELL_TELEMETRY_OPTOUT
1
>
```

after upgrade (and a full system reboot just to make sure), the environment optout is preserved:
```
> $env:POWERSHELL_TELEMETRY_OPTOUT
1
>
```

so i think i can finally mark this issue as closed/fixed.
